src/web/history.rkt

```diff
@@ -147,3 +147,107 @@
+             (type . "regimes") 
+             (conditions . ,(for/list ([entry prevs] [idx (in-naturals)]) 
+                (let ([entry-ivals (filter (λ (intrvl) (= (interval-alt-idx intrvl) idx)) intervals)]) (map (curryr interval->string repr) entry-ivals))))
+             (prev . ,(for/list ([entry prevs] 
+                      [new-pcontext (split-pcontext pcontext splitpoints prevs ctx)]
+                      [new-pcontext2 (split-pcontext pcontext2 splitpoints prevs ctx)])
+               (render-json entry new-pcontext new-pcontext2 ctx)))
+                   )]
+
+    [(alt prog `(taylor ,loc ,pt ,var) `(,prev))
+     `#hash(
+             (program . ,(fpcore->string (program->fpcore prog ctx))) 
+             (type . "taylor") 
+             (prev . ,(render-json prev pcontext pcontext2 ctx))
+             (pt . ,(~a pt))
+             (var . ,(~a var))
+             (loc . ,loc)
+             (error . ,err) 
+             (training-error . ,err2))]
+
+    [(alt prog `(simplify ,loc ,input ,proof ,soundiness) `(,prev))
+     `#hash(
+             (program . ,(fpcore->string (program->fpcore prog ctx))) 
+             (type . "simplify") 
+             (prev . ,(render-json prev pcontext pcontext2 ctx))
+             (proof . ,(if proof (render-proof-json proof soundiness pcontext ctx) ""))
+             (loc . ,loc)
+             (error . ,err) 
+             (training-error . ,err2))]
+
+    [(alt prog `initial-simplify `(,prev))
+     `#hash(
+             (program . ,(fpcore->string (program->fpcore prog ctx))) 
+             (type . "initial-simplify") 
+             (prev . ,(render-json prev pcontext pcontext2 ctx))
+             (error . ,err) 
+             (training-error . ,err2))]
+
+    [(alt prog `final-simplify `(,prev))
+     `#hash(
+             (program . ,(fpcore->string (program->fpcore prog ctx))) 
+             (type . "final-simplify") 
+             (prev . ,(render-json prev pcontext pcontext2 ctx))
+             (error . ,err) 
+             (training-error . ,err2))]
+
+    [(alt prog `(rr ,loc ,input ,proof ,soundiness) `(,prev))
+     `#hash(
+             (program . ,(fpcore->string (program->fpcore prog ctx))) 
+             (type . "rr") 
+             (prev . ,(render-json prev pcontext pcontext2 ctx))
+             (proof . ,(if proof (render-proof-json proof soundiness pcontext ctx) ""))
+             (rule . ,(if (rule? input) "rewrite-once" "egg-rr"))
+             (loc . ,loc)
+             (error . ,err) 
+             (training-error . ,err2))]))
+
+(define (render-proof-json proof soundiness pcontext ctx)
+      (for/list ([step proof] [sound soundiness])
+          (define-values (dir rule loc expr) (splice-proof-step step))
+          (define err
+            (format-accuracy (errors-score (errors expr pcontext ctx))
+                             (representation-total-bits (context-repr ctx))
+                             #:unit "%"))
```

Comment:
Again here too I would recommend doing the errors as floats not as a formatted string.